File: Lab Report 1/BankAccount(Task-3).cpp
#include<iostream>
using namespace std;

class BankAccount
{
private:
    int balance;
public:
    BankAccount(){
        balance = 0;
    }

    BankAccount(int amount){
        balance = amount;
    }

    void deposit(int amount){
        balance += amount;
    }

    void withdraw(int amount){
        if(balance <= 500)
            cout<<"Insufficient Balance!"<<endl;
        else
            balance -= amount;
    }

    void display(){
        cout<<"Current Balance: "<<balance<<endl;
    }
};

main()
{
    BankAccount Acc1(1000);

    Acc1.display();
    Acc1.deposit(500);
    Acc1.display();
    Acc1.withdraw(500);
    Acc1.display();
    Acc1.withdraw(500);
    Acc1.display();
    Acc1.withdraw(500);
    Acc1.display();
}

File: Lab Report 1/Python/BankAccount(Task 3).py
class BankAccount:
    balance = 1000

    def __init__(self):
        balance = 1000

    def deposit(self, amount):
        self.balance += amount

    def withdraw(self, amount):
        if self.balance <= 500:
            print("Insufficient Balance")
        else:
            self.balance -= amount

    def display(self):
        print("Current Balance : ", self.balance)


Acc1 = BankAccount()

Acc1.display()
Acc1.deposit(500)
Acc1.display()
Acc1.withdraw(500)
Acc1.display()
Acc1.withdraw(500)
Acc1.display()
Acc1.withdraw(500)
Acc1.display()